src/httpx_utils.py

```
@@ -0,0 +1,42 @@
+    """Return True when the FORCE_IPV4 env var is set to a truthy value."""
+    return os.environ.get("FORCE_IPV4", "").lower() in ("1", "true", "yes")
+
+
+def get_async_transport(**kwargs) -> httpx.AsyncHTTPTransport | None:
+    """Return an IPv4-bound async transport when FORCE_IPV4 is enabled.
+
+    Pass the result as ``transport=`` to :class:`httpx.AsyncClient`.
+    Returns ``None`` when no override is needed (the default).
+    """
+    if _force_ipv4_enabled():
+        return httpx.AsyncHTTPTransport(local_address="0.0.0.0", **kwargs)
```

Ruff (0.15.11)

[error] Possible binding to all interfaces

(S104)